routing/http/server/server.go

```diff
@@ -495,21 +503,28 @@ func writeResultsIterNDJSON[T any](w http.ResponseWriter, resultIter iter.Result
 	defer resultIter.Close()
 
 	w.Header().Set("Content-Type", mediaTypeNDJSON)
-	w.WriteHeader(http.StatusOK)
```

Comment:
ℹ️ This is automatically added by Go once we start writing the response. This is good so that we can actually set the Cache-Control header before writing the first thing.